app/models/pet.rb

```diff
@@ -10,4 +13,12 @@ def shelter_name
   def self.adoptable
     where(adoptable: true)
   end
+
+  def self.adoptable_search(search_params)
+    adoptable.search(search_params)
+  end
+
+  def update_adoptable_status
+    update(adoptable: false) if pet_applications.any? { |pa| pa.approval == true }
```

Comment:
These are okay model methods to put some Ruby logic in (as you've done) because there isn't an easier way to do this in AR